### PR TITLE
added q value parameter to be passed to filter functions

### PR DESCRIPTION
**Pull request by willxwmusic**
I have added a 2nd parameter to the filter functions low_pass, high_pass, to_low_pass, and to_high pass; allowing a q value to be passed along with the frequency.

### CODE REVIEW

**Comment by willxwmusic** [comment]
this is intended as a solution to #514

**Comment by willxwmusic** [comment]
I agree, this is a better solution.
I will update the proposed API to involve an option defaulting to 0.5, the original default q value within new methods such as `high_pass_with_q()`.
Thank you for your help.

**Comment by willxwmusic** [comment]
I have added secondary methods which are the same as `low_pass`, `high_pass`, `to_low_pass` and `to_high_pass` titled `low_pass_with_q`, etc.
They allow the q value to be passed to these functions too, changing the bandwidth of said filters.

**Comment by dvdsk** [comment]
> Creating a new method in order to perform exactly the same function but with an extra parameter is not an efficient solution to the issue.

I think I have got a good solution for that. Lets make the implementation of the non q versions call the `with_q` versions with as q value  `0.5`. The compiler will inline the call so this won't make the code slower.

**Comment by willxwmusic** [comment]
I have implemented this correction. `low_pass()`, etc. now just call `low_pass_with_q()`,etc. and pass the cutoff frequency along with the original default Q value of 0.5. This means that previously written code will not be broken and downstream users will not have to update their code, however, the option to change the q value is now there if required.

**Comment by willxwmusic** [comment]
I believe this could be better implemented using a default instead of a secondary method. I will implement this change

**Comment by willxwmusic** [comment]
Upon revision. that would not be possible. I believe this should be pulled to master as the `with_q` functions work as desired.

File: src/source/blt.rs [hunk 1]
```diff
@@ -21,13 +36,14 @@ where
     }
 }
 
-pub fn high_pass<I>(input: I, freq: u32) -> BltFilter<I>
+/// Same as low_pass but allows the q value (bandwidth) to be changed
```
Comment (dvdsk) [inline src/source/blt.rs:39]:
```suggestion
/// Same as high_pass but allows the q value (bandwidth) to be changed
```

File: src/source/mod.rs [hunk 2]
```diff
@@ -351,6 +351,23 @@ where
     {
         blt::high_pass(self, freq)
     }
+
```
Comment (dvdsk) [inline src/source/mod.rs:354]: Think you forgot the #[inline] here

```suggestion

#[inline]
```

File: src/source/mod.rs [hunk 3]
```diff
@@ -351,6 +351,23 @@ where
...
+    fn low_pass_with_q(self, freq: u32, q: f32) -> BltFilter<Self>
+    where
+        Self: Sized,
+        Self: Source<Item = f32>,
+    {
+        blt::low_pass_with_q(self, freq, q)
+    }
+
```
Comment (dvdsk) [inline src/source/mod.rs:362]: Lets add a comment explaining what q is like you have in blt.rs

**Review by dvdsk — CHANGES_REQUESTED** [review]
Looks great, I found some small stuff but that should be easy to fix. 

After that you should ask est31 for a review. They are a maintainer and can merge this. I am only trying to help them out a bit.

File: src/source/mod.rs (hunk 3 above)
Comment (willxwmusic) [reply src/source/mod.rs:365]: Added comments similar to that of blt.rs

**Comment by willxwmusic** [comment]
My apologies for the rookie mistakes. I am fairly new to rust as a programming language. I will submit the pull request for review.

**Comment by willxwmusic** [comment]
@est31 could this please be merged? Thanks.

**Review by est31 — APPROVED** [review]

**Pull request merged**